tests/implementations/register.test.ts

```diff
@@ -0,0 +1,60 @@
+import EnvRegistry from '../../src/EnvRegistry';
+import { environments } from './mocks';
+import { validationMessages } from '../../src/utils';
+
+describe('Register test suite', () => {
+  let envRegistry: EnvRegistry;
+
+  beforeEach(() => {
+    localStorage.clear();
+    envRegistry = new EnvRegistry('token');
+  });
+
+  it('Calling register method registers the version of the provided envKey and Env ', async (done) => {
+    expect.assertions(1);
+    await envRegistry.register({ envKey: 'master', env: environments.master});
+    envRegistry.environments$({}).subscribe((env) => {
+      expect(env).toEqual({key: 'master', value: environments.master });
```

Comment:
As I understand, the stream will complete after this emition. Suggest to put done in the complete callback